### DISCUSSION TITLE
AvailableSize with winit

### DISCUSSION THREAD

**Discussion by DasLixou** [DioxusLabs/taffy#321] [Q&A] [answered]
Hey!
I'm creating a window with winit and now want to `compute_layout` of my taffy nodes with the size of the window. My question is:
should i use winit's `LogicalSize` or `PhysicalSize`?

And secondly:
I currently just pass
```rust
Size {
    width: AvailableSpace::Definite(1280.),
    height: AvailableSpace::Definite(720.),
}
```
but when i print the layout of my size=100% element its:
`Layout { order: 0, size: Size { width: 128000.0, height: 72000.0 }, location: Point { x: 0.0, y: 0.0 } }`
and i wonder how i can convert the `width: 128000.0, height: 72000.0` back to pixel data.

Thanks :D

**Comment by adjabaev** [comment, marked as answer]
> but when i print the layout of my size=100% element its:
Layout { order: 0, size: Size { width: 128000.0, height: 72000.0 }, location: Point { x: 0.0, y: 0.0 } }
and i wonder how i can convert the width: 128000.0, height: 72000.0 back to pixel data.

You probably should pass your percentage on a 0.0-1.0 scale

**Reply by DasLixou** [reply]
Oh LOL but that’s not how percentage works 😂 and I wondered why this is using a float

**Reply by DasLixou** [reply]
And what about the sizes?

**Comment by nicoburns** [comment]
> My question is: should i use winit's `LogicalSize` or `PhysicalSize`.

You can do either. Taffy doesn't care what your units are: whatever units you put it, you'll values in the same units out. So you can do any conversions to either the inputs to Taffy or the outputs from Taffy. One thing to be aware of is that Taffy will round the layout to exact pixel sizes, so if you have a preferred unit to do rounding in then you should use that.

**Reply by DasLixou** [reply]
Ok Thanks a lot :D
